salt/modules/win_shortcut.py

```diff
@@ -0,0 +1,471 @@
+"""
+Execution module for creating shortcuts on Windows. Handles file shortcuts
+(`.lnk`) and url shortcuts (`.url`). Allows for the configuration of icons and
+hot keys on file shortcuts. Changing the icon and hot keys are unsupported for
+url shortcuts.
+"""
+# https://docs.microsoft.com/en-us/troubleshoot/windows-client/admin-development/create-desktop-shortcut-with-wsh
+# https://docs.microsoft.com/en-us/previous-versions/windows/internet-explorer/ie-developer/windows-scripting/f5y78918(v=vs.84)
+import logging
+import os
+import time
+
+import salt.utils.path
+import salt.utils.platform
+import salt.utils.winapi
+from salt.exceptions import CommandExecutionError
+
+HAS_WIN32 = False
+if salt.utils.platform.is_windows():
+    import win32com.client
+
+    HAS_WIN32 = True
+
+# Set up logging
+log = logging.getLogger(__name__)
+
+# Define the module's virtual name
+__virtualname__ = "shortcut"
+
+
+WINDOW_STYLE = {
+    1: "Normal",
+    3: "Maximized",
+    7: "Minimized",
+    "Normal": 1,
+    "Maximized": 3,
+    "Minimized": 7,
+}
+
+
+def __virtual__():
+    """
+    Make sure we're on Windows
+    """
+    # Verify Windows
+    if not salt.utils.platform.is_windows():
+        log.debug("Shortcut module only available on Windows systems")
+        return False, "Shortcut module only available on Windows systems"
+    if not HAS_WIN32:
+        log.debug("Shortcut module requires pywin32")
+        return False, "Shortcut module requires pywin32"
+
+    return __virtualname__
+
+
+def get(path):
+    r"""
+    Gets the properties for a shortcut
+
+    Args:
+        path (str): The path to the shortcut
+
+    Returns:
+        dict: A dictionary containing all available properties for the specified
+            shortcut
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt * shortcut.get path="C:\path\to\shortcut.lnk"
+    ..
+    """
+    if not os.path.exists(path):
```

Comment:
Can we also verify its a shortcut?

Reply:
Now makes sure the path has a `.lnk` or `.url` file extension.